modules/renter/contractor/allowance.go

```diff
@@ -0,0 +1,161 @@
+package contractor
+
+import (
+	"errors"
+
+	"github.com/NebulousLabs/Sia/modules"
+	"github.com/NebulousLabs/Sia/types"
+)
+
+// SetAllowance sets the amount of money the Contractor is allowed to spend on
+// contracts over a given time period, divided among the number of hosts
+// specified. Note that Contractor can start forming contracts as soon as
+// SetAllowance is called; that is, it may block.
+//
+// In most cases, SetAllowance will renew existing contracts instead of
+// forming new ones. This preserves the data on those hosts. When this occurs,
+// the renewed contracts will atomically replace their previous versions. If
+// SetAllowance is interrupted, renewed contracts may be lost, though the
+// allocated funds will eventually be returned,
+//
+// TODO: can an Editor or Downloader be used across renewals?
+// TODO: will hosts allow renewing the same contract twice?
+//
+// NOTE: At this time, transaction fees are not counted towards the allowance.
+// This means the contractor may spend more than allowance.Funds.
+func (c *Contractor) SetAllowance(a modules.Allowance) error {
+	// sanity checks
+	if a.Hosts == 0 {
+		return errors.New("hosts must be non-zero")
+	} else if a.Period == 0 {
+		return errors.New("period must be non-zero")
+	} else if a.RenewWindow == 0 {
+		return errors.New("renew window must be non-zero")
+	} else if a.RenewWindow >= a.Period {
+		return errors.New("renew window must be less than period")
+	}
+
+	// check that allowance is sufficient to store at least one sector
+	numSectors, err := maxSectors(a, c.hdb)
+	if err != nil {
+		return err
+	} else if numSectors == 0 {
+		return errInsufficientAllowance
+	}
+
+	c.mu.RLock()
+	shouldRenew := a.Period != c.allowance.Period || a.Funds.Cmp(c.allowance.Funds) != 0
```

Comment:
Best to verify your assumptions by writing a test. It's easy for other developers to not realize they are assumptions and break things if there's no test to inform them that they've broken compatibility.

I'm pretty sure you are correct though, hosts will reject any renewals that don't have a strictly greater height (until your change). I'm not 100% sure why this is the case.